domain/src/main/java/edu/nextstep/camp/calculator/Calculator.kt

```diff
@@ -1,47 +1,21 @@
 package edu.nextstep.camp.calculator
 
-import java.util.Stack
-
 class Calculator {
 
     fun evaluate(expression: String?): Int {
-        val stack = Stack<String>()
+        val expressionStack = ExpressionStack()
         val expressionWithoutBlank = expression?.split(DELIMITER) ?: throw IllegalArgumentException()
 
-        expressionWithoutBlank.forEach {
-            if (stack.isEmpty()) {
-                stack.push(it)
-            } else {
-                pushValueWhenStackIsNotEmpty(stack, it)
+        expressionWithoutBlank.forEach { str ->
+            val stack = expressionStack.getStackForCalculating(str)
+            stack?.let {
+                val type = it.pop()
+                val value = it.pop()
+                val result = calculateValue(type, value.toInt(), str.toInt())
+                expressionStack.pushResult(result.toString())
```

Comment:
getStackForCalculating 함수가 내부 Stack에 푸시하는 코드와 실제 함수 이름 처럼 Stack을 가져오는 기능이 혼재하고 있습니다. null 이면 내부적으로 stack에 푸시한것이고 stack이 null이 아니면 가져온 스택으로 stack?.let 블럭에서 계산을 합니다.

```suggestion
            // str에 대한 입력만을 담당하는 함수
            expressionStack.push(str)
            
            // 현재 계산 가능한 상태인지 체크하는 함수
            if (expressionStack.isReadyForCalculating()) { 
                // 계산을 진행
                val type = expressionStack.getType()
                val value = expressionStack.getValue()
                val result = calculateValue(type, value.toInt(), str.toInt())
                expressionStack.pushResult(result.toString())
            }
```

동작하는 코드라기보단 함수 분리를 이렇게 하면 어떨까하는 코드로만 이해해 주시기 바랍니다.
null로 어떤 로직을 판단하기 보다는 명확한 API로 판단하는게 좋은 코드라고 생각합니다. ✌